### PR TITLE
Upload to bucket

### PR DESCRIPTION
**Pull request by koltoi**

### CODE REVIEW

File: consul-backup/main.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		log.Error(err, "get client error")
+	} else {
+		backupStorage.BackupCRStat = &backupStorage.BackupCRStatus{Client: k8sclient}
+		log.Infof("before upload")
+		backupStorage.BackupCRStat.UploadDataToBucket(watchNamespace)
+	}
+
+	for {
+		log.Infof("sleeping...")
+		time.Sleep(1*time.Hour)
```
[the remark this answers is not in the document]
Comment (koltoi) [reply consul-backup/main.go:46]: add config map to make the time period configurable

File: consul-backup/pkg/s3client/s3client.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	}
+
+	err = file.Sync()
+	if err != nil {
+		log.Error(err, "Failed to sync file")
+		return err
+	}
+
+	// In this function a file is uploaded with the same name to the bucket.
+	// Backup-operator uses versioning-enabled buckets, that allows to keep multiple variants of an object in the same bucket.
```
Comment (koltoi) [inline consul-backup/pkg/s3client/s3client.go:60]: no oversioning

File: consul-backup/pkg/s3client/s3client.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	if err != nil {
+		log.Error(err, "Failed to put file into bucket")
+		return err
+	}
+	log.Info("--------client successfully", "uploaded object:", uploadInfo)
+
+	return nil
+}
+
+/*
```
Comment (koltoi) [inline consul-backup/pkg/s3client/s3client.go:73]: remove

File: consul-operator/deployment/app-deployment/templates/consul-ss.yaml [hunk 4]
```diff
@@ -90,3 +90,21 @@ spec:
               name: consuldns
             - containerPort: {{.Values.service.server}}
               name: server
+        - name: consul-backup
+          image: registry.dac.nokia.com/public/backup:1.0.0
```
Comment (koltoi) [inline consul-operator/deployment/app-deployment/templates/consul-ss.yaml:94]: consul-backup

File: consul-backup/main.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	err = serviceconfig.ReadServiceConfig()
+	if err != nil {
+		log.Error(err, "Failed to read configmap ")
+		panic(err)
+	}
+
+	err = backupService.StartPeriodicBackup(ownNamespace)
+	err = serviceconfig.ReadServiceConfig()
```
Comment (hochdorf) [inline consul-backup/main.go:32]: is this second readServiceConfig needed?

**Review by hochdorf — CHANGES_REQUESTED** [review]
pls resolve that last review comment

**Review by hochdorf — APPROVED** [review]

**Pull request merged**